src/sources/aws_kinesis_firehose/mod.rs

```diff
@@ -50,6 +50,13 @@ pub struct AwsKinesisFirehoseConfig {
     #[configurable(metadata(docs::examples = "access_keys_example()"))]
     access_keys: Option<Vec<SensitiveString>>,
 
+    /// Whether or not to forward the access key with events.
+    ///
+    /// If set to `true`, when incoming requests contains an access key, it will be kept in the
+    /// event metadata.
```

Comment:
We failed to document _what_ the secret name is for the other two places event secrets are actually used, but we should make sure to do that here.

I'd like to clarify this as an `AWS Access Key` as well.
```suggestion
    /// Whether or not to store the AWS Access Key in event secrets.
    ///
    /// If set to `true`, when incoming requests contains an access key it will be kept in the
    /// event secrets.
```

I'm iffy on the phrase "event secrets", but we think it's important to clarify it's separate from the general "metadata".

Reply:
I'd say `AWS Firehose Access Key` to avoid confusion with IAM access keys.